submissions/kissik/data-structures/index.js

```diff
@@ -0,0 +1,110 @@
+const http = require('http');
+
+const fs = require('fs');
+
+const port = process.env.PORT || 3210;
+
+const list = {};
+
+list.data = {};
+
+list.push = data => {
+  list.data = {
+    item: data,
+    next: list.data
+  };
+};
+
+list.pop = () => {
+  const a = list.data.item ? list.data.item : 'none';
+  if (list.data) list.data = list.data.next;
+  return a;
+};
+
+list.print = msg => {
+  let a = list.data;
+  let str = '';
+  str += `\nThe ${msg}:\n`;
+  while (a && a.item) {
+    str += `${a.item} -> `;
+    a = a.next;
+  }
+  return str;
+};
+
+list.remove = data => {
+  let a = list.data;
+  let next;
+  if (a.next) next = a.next;
+  if (a.item === data) list.data = a.next;
+  while (next && next.item !== data) {
+    next = next.next;
+    a = a.next;
+  }
+  if (next && next.item === data) a.next = next.next;
+};
+
+http
+  .createServer((req, res) => {
```

Comment:
It is veeery long function with a bunch of repeated code.

Reply:
done